packages/deployments/contracts/contracts_forge/RelayerFeeRouter.t.sol

```diff
@@ -0,0 +1,205 @@
+
+// other forge commands: yarn workspace @connext/nxtp-contracts forge <CMD>
+// see docs here: https://onbjerg.github.io/foundry-book/index.html
+
+contract MockHome {
+  function dispatch(
+    uint32 _destinationDomain,
+    bytes32 _recipientAddress,
+    bytes memory _messageBody
+  ) external {
+    1 == 1;
+  }
+}
+
+contract RelayerFeeRouterTest is ForgeHelper {
+  using TypedMemView for bytes;
+  using TypedMemView for bytes29;
+  using RelayerFeeMessage for bytes29;
+
+  // ============ Libraries ============
+  using stdStorage for StdStorage;
+
+  event Send(
+    uint32 domain,
+    address recipient,
+    uint256 amount,
+    bytes32[] _transactionIds,
+    bytes32 remote,
+    bytes message
+  );
+
+  event Receive(
+    uint64 indexed originAndNonce,
+    uint32 indexed origin,
+    address indexed recipient,
+    uint256 amount,
+    bytes32[] transactionIds
+  );
+
+  event SetConnext(address indexed connext);
+
+  // ============ Storage ============
+
+  RelayerFeeRouter relayerFeeRouter;
+  RelayerFeeRouter relayerFeeRouterImplementation;
+  ERC1967Proxy proxy;
+
+  uint256 domain = 1;
+  address bridgeRouter = address(1);
+  address tokenRegistry = address(2);
+  address wrapper = address(3);
+
+  address internal xAppConnectionManager = address(1);
+  address internal home;
+  address internal connext = address(this);
+  address internal connext2 = address(3);
+  address internal recipient = address(4);
+  bytes32 internal remote = "remote";
+  uint32 internal localDomain = uint32(123);
+  uint32 internal remoteDomain = uint32(1);
+
+  // ============ Test set up ============
+
+  function setUp() public {
+    home = address(new MockHome());
+    vm.mockCall(xAppConnectionManager, abi.encodeWithSignature("home()"), abi.encode(home));
+    vm.mockCall(xAppConnectionManager, abi.encodeWithSignature("isReplica(address)"), abi.encode(bool(true)));
+    // vm.mockCall(home, abi.encodeWithSelector(Home.dispatch.selector), bytes("0x"));
+    vm.mockCall(home, abi.encodeWithSignature("localDomain()"), abi.encode(localDomain));
+
+    relayerFeeRouterImplementation = new RelayerFeeRouter();
+
+    proxy = new ERC1967Proxy(
+      address(relayerFeeRouterImplementation),
+      abi.encodeWithSelector(RelayerFeeRouter.initialize.selector, xAppConnectionManager, connext)
+    );
+
+    relayerFeeRouter = RelayerFeeRouter(payable(address(proxy)));
+
+    // Shouldn't enrollRemoteRouter be protected by onlyOwner?
+    relayerFeeRouter.enrollRemoteRouter(remoteDomain, bytes32(remote));
+  }
+
+  // ============ initialize ============
+  function testInitializeParameters() public {
+    assertEq(address(relayerFeeRouter.xAppConnectionManager()), address(xAppConnectionManager));
+    assertEq(address(relayerFeeRouter.connext()), address(connext));
+  }
+
+  // ============ setConnext ============
+
+  // Should work
+  function testSetConnext() public {
+    vm.expectEmit(true, true, true, true);
+    emit SetConnext(connext2);
```

Comment:
Should add a similar event to the `BridgeRouter` (can be done via `ConnextRouter` contract)